### PR TITLE
OSD-12962 findOrCreateSecurityGroup unit test

### PR DESCRIPTION
**Pull request by mjlshen**
Pulled out `findOrCreateSecurityGroup` from `validateSecurityGroup` and standardized it with `findOrCreateVpcEndpoint`

### CODE REVIEW

File: controllers/vpcendpoint/helpers.go [hunk 1]
```diff
@@ -167,6 +227,12 @@ func (r *VpcEndpointReconciler) findOrCreateVpcEndpoint(resource *avov1alpha1.Vp
 		vpce = resp.VpcEndpoints[0]
```
Comment (dustman9000) [inline controllers/vpcendpoint/helpers.go:227]: this seems a little brittle....add a nill check?

File: controllers/vpcendpoint/helpers.go [hunk 2]
```diff
@@ -167,6 +227,12 @@ func (r *VpcEndpointReconciler) findOrCreateVpcEndpoint(resource *avov1alpha1.Vp
...
 	}
 
+	resource.Status.VPCEndpointId = *vpce.VpcEndpointId
```
Comment (dustman9000) [inline controllers/vpcendpoint/helpers.go:230]: Add a nil check for `vpce` above this deref?

Comment (mjlshen) [reply controllers/vpcendpoint/helpers.go:238]: Added! Same for when I similarly dereferenced the security group to get its ID.

File: controllers/vpcendpoint/helpers.go (hunk 1 above)
Comment (mjlshen) [reply controllers/vpcendpoint/helpers.go:231]: Talked about this and there should always be at least one VpcEndpoint if we get to this point due to a previous check for

```go
if resp == nil || len(resp.VpcEndpoints) == 0 {
```

**Comment by dustman9000** [comment]
thanks!
/lgtm

**Pull request merged**